Stop gsignal warnings on window-close

lukefromdc opened the pull request:
based on https://github.com/mate-desktop/atril/commit/fda33fbeedc0aab64d9479850047d0817d0b38be
"Check if handler_id is connected before disconnect" from Atril

lukefromdc commented:
Build tests with GTK3.16, 3.18, 3.20, and 3.22 all completed and the affected warnings were gone from each one in test runs. A GTK 2 build also completed without issue but could not be run over a GTK 3 system due to the usual mixed GTK2/GTK3 symbol issue. Will merge this unless someone finds a problem or other reason not to

lukefromdc commented:
Merged, Pushed from local branch to avold merge commit
https://github.com/mate-desktop/caja/commit/24bb72b321c044479728fd1d43c941afc0309f18

raveit65 commented:
hmm, you're faster than i can test it.
I was just testing GtkApplication PR again....

raveit65 commented:
Something goes wrong, after your last commit i get a conflict with GtkApplication branch.
```
[rave@mother caja]$ LANG=C
[rave@mother caja]$ git status
On branch dev-GtkApplication3
Your branch is up-to-date with 'origin/dev-GtkApplication3'.
nothing to commit, working directory clean
[rave@mother caja]$ git rebase master
First, rewinding head to replay your work on top of it...
Applying: GTK3: port libunique ->GtkApplication
Using index info to reconstruct a base tree...
M	configure.ac
M	data/caja.desktop.in.in
M	src/Makefile.am
M	src/caja-application.c
M	src/caja-application.h
M	src/caja-bookmarks-window.c
M	src/caja-connect-server-dialog-nonmain.c
M	src/caja-location-bar.c
M	src/caja-main.c
M	src/caja-navigation-window-menus.c
M	src/caja-navigation-window.c
M	src/caja-places-sidebar.c
M	src/caja-spatial-window.c
M	src/caja-window-manage-views.c
M	src/caja-window.c
.git/rebase-apply/patch:329: trailing whitespace.
#endif  
.git/rebase-apply/patch:438: trailing whitespace.
        /* Open windows at each requested location. */ 
.git/rebase-apply/patch:440: trailing whitespace.
        while (i < len ){ 
.git/rebase-apply/patch:547: trailing whitespace.
                              NULL); 
.git/rebase-apply/patch:754: trailing whitespace.
        /* Hold Caja open if the desktop is showing as autostart mode  
warning: squelched 23 whitespace errors
warning: 28 lines add whitespace errors.
Falling back to patching base and 3-way merge...
Auto-merging src/caja-window.c
CONFLICT (content): Merge conflict in src/caja-window.c
Auto-merging src/caja-window-manage-views.c
CONFLICT (content): Merge conflict in src/caja-window-manage-views.c
Auto-merging src/caja-spatial-window.c
CONFLICT (content): Merge conflict in src/caja-spatial-window.c
Auto-merging src/caja-places-sidebar.c
CONFLICT (content): Merge conflict in src/caja-places-sidebar.c
Auto-merging src/caja-navigation-window.c
CONFLICT (content): Merge conflict in src/caja-navigation-window.c
Auto-merging src/caja-navigation-window-menus.c
CONFLICT (content): Merge conflict in src/caja-navigation-window-menus.c
Auto-merging src/caja-main.c
CONFLICT (content): Merge conflict in src/caja-main.c
Auto-merging src/caja-connect-server-dialog-nonmain.c
CONFLICT (content): Merge conflict in src/caja-connect-server-dialog-nonmain.c
Auto-merging src/caja-bookmarks-window.c
CONFLICT (content): Merge conflict in src/caja-bookmarks-window.c
Auto-merging src/caja-application.h
CONFLICT (content): Merge conflict in src/caja-application.h
Auto-merging src/caja-application.c
CONFLICT (content): Merge conflict in src/caja-application.c
Auto-merging src/Makefile.am
CONFLICT (content): Merge conflict in src/Makefile.am
Auto-merging data/caja.desktop.in.in
Auto-merging configure.ac
CONFLICT (content): Merge conflict in configure.ac
error: Failed to merge in the changes.
Patch failed at 0001 GTK3: port libunique ->GtkApplication
The copy of the patch that failed is found in: .git/rebase-apply/patch

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".
```

Ok, to late for me....i can't merge GtkApplication branch this evenning.

lukefromdc commented:
I just merged it, had no trouble on this end

On 10/8/2016 at 3:48 PM, "raveit65" <notifications@github.com> wrote:
>
>Something goes wrong, after your last commit i get a conflict with 
>GtkApplication branch.
>```
>[rave@mother caja]$ LANG=C
>[rave@mother caja]$ git status
>On branch dev-GtkApplication3
>Your branch is up-to-date with 'origin/dev-GtkApplication3'.
>nothing to commit, working directory clean
>[rave@mother caja]$ git rebase master
>First, rewinding head to replay your work on top of it...
>Applying: GTK3: port libunique ->GtkApplication
>Using index info to reconstruct a base tree...
>M	configure.ac
>M	data/caja.desktop.in.in
>M	src/Makefile.am
>M	src/caja-application.c
>M	src/caja-application.h
>M	src/caja-bookmarks-window.c
>M	src/caja-connect-server-dialog-nonmain.c
>M	src/caja-location-bar.c
>M	src/caja-main.c
>M	src/caja-navigation-window-menus.c
>M	src/caja-navigation-window.c
>M	src/caja-places-sidebar.c
>M	src/caja-spatial-window.c
>M	src/caja-window-manage-views.c
>M	src/caja-window.c
>.git/rebase-apply/patch:329: trailing whitespace.
>#endif  
>.git/rebase-apply/patch:438: trailing whitespace.
>        /* Open windows at each requested location. */ 
>.git/rebase-apply/patch:440: trailing whitespace.
>        while (i < len ){ 
>.git/rebase-apply/patch:547: trailing whitespace.
>                              NULL); 
>.git/rebase-apply/patch:754: trailing whitespace.
>        /* Hold Caja open if the desktop is showing as autostart 
>mode  
>warning: squelched 23 whitespace errors
>warning: 28 lines add whitespace errors.
>Falling back to patching base and 3-way merge...
>Auto-merging src/caja-window.c
>CONFLICT (content): Merge conflict in src/caja-window.c
>Auto-merging src/caja-window-manage-views.c
>CONFLICT (content): Merge conflict in src/caja-window-manage-
>views.c
>Auto-merging src/caja-spatial-window.c
>CONFLICT (content): Merge conflict in src/caja-spatial-window.c
>Auto-merging src/caja-places-sidebar.c
>CONFLICT (content): Merge conflict in src/caja-places-sidebar.c
>Auto-merging src/caja-navigation-window.c
>CONFLICT (content): Merge conflict in src/caja-navigation-window.c
>Auto-merging src/caja-navigation-window-menus.c
>CONFLICT (content): Merge conflict in src/caja-navigation-window-
>menus.c
>Auto-merging src/caja-main.c
>CONFLICT (content): Merge conflict in src/caja-main.c
>Auto-merging src/caja-connect-server-dialog-nonmain.c
>CONFLICT (content): Merge conflict in src/caja-connect-server-
>dialog-nonmain.c
>Auto-merging src/caja-bookmarks-window.c
>CONFLICT (content): Merge conflict in src/caja-bookmarks-window.c
>Auto-merging src/caja-application.h
>CONFLICT (content): Merge conflict in src/caja-application.h
>Auto-merging src/caja-application.c
>CONFLICT (content): Merge conflict in src/caja-application.c
>Auto-merging src/Makefile.am
>CONFLICT (content): Merge conflict in src/Makefile.am
>Auto-merging data/caja.desktop.in.in
>Auto-merging configure.ac
>CONFLICT (content): Merge conflict in configure.ac
>error: Failed to merge in the changes.
>Patch failed at 0001 GTK3: port libunique ->GtkApplication
>The copy of the patch that failed is found in: .git/rebase-
>apply/patch
>
>When you have resolved this problem, run "git rebase --continue".
>If you prefer to skip this patch, run "git rebase --skip" instead.
>To check out the original branch and stop rebasing, run "git 
>rebase --abort".
>```
>
>Ok, to late for me....i can't merge GtkApplication branch this 
>evenning.
>
>-- 
>You are receiving this because you modified the open/close state.
>Reply to this email directly or view it on GitHub:
>https://github.com/mate-desktop/caja/pull/650#issuecomment-
>252444548

lukefromdc commented:
In merging GtkApplication3 I squashed all the commits and cleaned up
the commit message so it would not dominate the changelog

On 10/8/2016 at 3:48 PM, "raveit65" <notifications@github.com> wrote:
>
>Something goes wrong, after your last commit i get a conflict with 
>GtkApplication branch.
>```
>[rave@mother caja]$ LANG=C
>[rave@mother caja]$ git status
>On branch dev-GtkApplication3
>Your branch is up-to-date with 'origin/dev-GtkApplication3'.
>nothing to commit, working directory clean
>[rave@mother caja]$ git rebase master
>First, rewinding head to replay your work on top of it...
>Applying: GTK3: port libunique ->GtkApplication
>Using index info to reconstruct a base tree...
>M	configure.ac
>M	data/caja.desktop.in.in
>M	src/Makefile.am
>M	src/caja-application.c
>M	src/caja-application.h
>M	src/caja-bookmarks-window.c
>M	src/caja-connect-server-dialog-nonmain.c
>M	src/caja-location-bar.c
>M	src/caja-main.c
>M	src/caja-navigation-window-menus.c
>M	src/caja-navigation-window.c
>M	src/caja-places-sidebar.c
>M	src/caja-spatial-window.c
>M	src/caja-window-manage-views.c
>M	src/caja-window.c
>.git/rebase-apply/patch:329: trailing whitespace.
>#endif  
>.git/rebase-apply/patch:438: trailing whitespace.
>        /* Open windows at each requested location. */ 
>.git/rebase-apply/patch:440: trailing whitespace.
>        while (i < len ){ 
>.git/rebase-apply/patch:547: trailing whitespace.
>                              NULL); 
>.git/rebase-apply/patch:754: trailing whitespace.
>        /* Hold Caja open if the desktop is showing as autostart 
>mode  
>warning: squelched 23 whitespace errors
>warning: 28 lines add whitespace errors.
>Falling back to patching base and 3-way merge...
>Auto-merging src/caja-window.c
>CONFLICT (content): Merge conflict in src/caja-window.c
>Auto-merging src/caja-window-manage-views.c
>CONFLICT (content): Merge conflict in src/caja-window-manage-
>views.c
>Auto-merging src/caja-spatial-window.c
>CONFLICT (content): Merge conflict in src/caja-spatial-window.c
>Auto-merging src/caja-places-sidebar.c
>CONFLICT (content): Merge conflict in src/caja-places-sidebar.c
>Auto-merging src/caja-navigation-window.c
>CONFLICT (content): Merge conflict in src/caja-navigation-window.c
>Auto-merging src/caja-navigation-window-menus.c
>CONFLICT (content): Merge conflict in src/caja-navigation-window-
>menus.c
>Auto-merging src/caja-main.c
>CONFLICT (content): Merge conflict in src/caja-main.c
>Auto-merging src/caja-connect-server-dialog-nonmain.c
>CONFLICT (content): Merge conflict in src/caja-connect-server-
>dialog-nonmain.c
>Auto-merging src/caja-bookmarks-window.c
>CONFLICT (content): Merge conflict in src/caja-bookmarks-window.c
>Auto-merging src/caja-application.h
>CONFLICT (content): Merge conflict in src/caja-application.h
>Auto-merging src/caja-application.c
>CONFLICT (content): Merge conflict in src/caja-application.c
>Auto-merging src/Makefile.am
>CONFLICT (content): Merge conflict in src/Makefile.am
>Auto-merging data/caja.desktop.in.in
>Auto-merging configure.ac
>CONFLICT (content): Merge conflict in configure.ac
>error: Failed to merge in the changes.
>Patch failed at 0001 GTK3: port libunique ->GtkApplication
>The copy of the patch that failed is found in: .git/rebase-
>apply/patch
>
>When you have resolved this problem, run "git rebase --continue".
>If you prefer to skip this patch, run "git rebase --skip" instead.
>To check out the original branch and stop rebasing, run "git 
>rebase --abort".
>```
>
>Ok, to late for me....i can't merge GtkApplication branch this 
>evenning.
>
>-- 
>You are receiving this because you modified the open/close state.
>Reply to this email directly or view it on GitHub:
>https://github.com/mate-desktop/caja/pull/650#issuecomment-
>252444548

raveit65 commented:
I suggest to rebase master and reverting your last commit.
To get the repo clean again. I will do that tomorrow.
> Merged, Pushed from local branch to avold merge commit

This was a hint that something is wrong.

lukefromdc commented:
The reason that one tried to generate a merge commit when I first tried a 
local merge was that there were multiple commits in the branch. When I
squashed them it merged clean locally, I then pushed from that.

On 10/8/2016 at 3:59 PM, "raveit65" <notifications@github.com> wrote:
>
>I suggest to rebase master and reverting your last commit.
>To get the repo clean again. I will do that tomorrow.
>> Merged, Pushed from local branch to avold merge commit
>
>This was a hint that something is wrong.
>
>-- 
>You are receiving this because you modified the open/close state.
>Reply to this email directly or view it on GitHub:
>https://github.com/mate-desktop/caja/pull/650#issuecomment-
>252445100

lukefromdc commented:
If you revert and reapply over repo issues, you might want to first cut and
paste that cleaned-up commit message from GtkApplication3 so you can
avoid having to edit that all over again when reapplying the merge.

On 10/8/2016 at 3:59 PM, "raveit65" <notifications@github.com> wrote:
>
>I suggest to rebase master and reverting your last commit.
>To get the repo clean again. I will do that tomorrow.
>> Merged, Pushed from local branch to avold merge commit
>
>This was a hint that something is wrong.
>
>-- 
>You are receiving this because you modified the open/close state.
>Reply to this email directly or view it on GitHub:
>https://github.com/mate-desktop/caja/pull/650#issuecomment-
>252445100

raveit65 commented:
> In merging GtkApplication3 I squashed all the commits and cleaned up the commit message so it would not dominate the changelog

This explains a lot.
Looks like testing it again in f24 and f25 was quite useless the last 2 hours.
Thank you

lukefromdc commented:
Sorry about the hassle, maybe you tried to merge after it had already
been merged?  If you find  problems don't hesitate to revert, I left
the branch on the repo

On 10/8/2016 at 4:10 PM, "raveit65" <notifications@github.com> wrote:
>
>> In merging GtkApplication3 I squashed all the commits and 
>cleaned up the commit message so it would not dominate the 
>changelog
>
>This explains a lot.
>Looks like testing it again in f24 and f25 was quite useless the 
>last 2 hours.
>Thank you
>
>-- 
>You are receiving this because you modified the open/close state.
>Reply to this email directly or view it on GitHub:
>https://github.com/mate-desktop/caja/pull/650#issuecomment-
>252445765

raveit65 commented:
>  maybe you tried to merge after it had already been merged?

Yes, this was the problem and you did not push the squash to the GtkApplication3 branch.
So i can only delete the branch local to get the repo clean again.
Any ways, i tested the branch again before you squashed it, so it have to be equal.
Personal i hadn't squash the commits for the desktop files in one commit, but on the other side it's better for back porting your work to 1.16 or using it as patch for 1.16 in debian.

raveit65 commented:
I did squash the branch local and there is no different between the resulting commit and you commit in master.
So all is good :)
Next time we should avoid to work at the same things. I will notice this in PR when i start to review and merge.

lukefromdc commented:
The reason I did not push the squashed version back to the GtkApplication3
branch was I wanted to keep it possible for others to bisect it if anything
really nasty came up and I couldn't find it myself. Otherwise I would have
deleted the branch after the push to master.

On 10/9/2016 at 2:00 AM, "raveit65" <notifications@github.com> wrote:
>
>>  maybe you tried to merge after it had already been merged?
>
>Yes, this was the problem and you did not push the squash to the 
>GtkApplication3 branch.
>So i can only delete the branch local to get the repo clean again.
>Any ways, i tested the branch again before you squashed it, so it 
>have to be equal.
>Personal i hadn't squash the commits for the desktop files in one 
>commit, but on the other side it's better for back porting your 
>work to 1.16 or using it as patch for 1.16 in debian.
>
>-- 
>You are receiving this because you modified the open/close state.
>Reply to this email directly or view it on GitHub:
>https://github.com/mate-desktop/caja/pull/650#issuecomment-
>252466909
